deserializers/src/main/java/com/nextdoor/bender/deserializer/json/GenericJsonDeserializer.java

```diff
@@ -36,21 +41,49 @@
   protected JsonParser parser;
   private final List<FieldConfig> nestedFieldConfigs;
   private String rootNodeOverridePath;
+  private final boolean performBase64DecodeAndUnzip;
 
   public GenericJsonDeserializer(List<FieldConfig> nestedFieldConfigs) {
-    this(nestedFieldConfigs, null);
+    this(nestedFieldConfigs, null, false);
   }
 
   public GenericJsonDeserializer(List<FieldConfig> nestedFieldConfigs,
-      String rootNodeOverridePath) {
+                                 String rootNodeOverridePath,
+                                 boolean performBase64DecodeAndUnzip) {
     this.nestedFieldConfigs = nestedFieldConfigs;
     this.rootNodeOverridePath = rootNodeOverridePath;
+    this.performBase64DecodeAndUnzip = performBase64DecodeAndUnzip;
+  }
+
+  public byte[] readGzipCompressedData(byte[] data) throws IOException {
+    GZIPInputStream gzipInputStream = new GZIPInputStream(new ByteArrayInputStream(data));
```

Comment:
Since we are streaming data I think we can re-use the `GZIPInputStream` in order to reduce the memory overhead. `GZIPInputStream` is buffered and I'd like to avoid creating it every time we need to decompress an event.

Reply:
GZIPInputStream doesn't support the reset() method, it throws an exception. There is a way around it to extend the class so that we reset the Inflater and pass in new input to it but don't think that's worth the effort right now.

Reply:
Okay, see if someone else has implemented this in a library.

Reply:
I don't see an open source library, we'd need to extend the class ourselves to get this functionality. There's an [example](http://tutorials.jenkov.com/java-io/bufferedinputstream.html#reusable-bufferedinputstream) here of how someone made a ReusableBufferedInputStream that's a good example.

So what we could do is have an additional method like setInputSource that just takes in a new input stream. I'm not sure whether this will get us any meaningful performance impact since a new input stream is needed anyways. I don't know if reusing input streams works since we need to reinstantiate for different files that come in, it's a different case than reusing the same input stream if you need the same content like [here](https://stackoverflow.com/questions/36623947/java-i-o-reuse-inputstream-object).